src/viam/module/module.py

```diff
@@ -75,14 +75,17 @@ async def _get_dependencies(self, dependencies: Sequence[str]) -> Mapping[Resour
 
     async def start(self):
         """Start the module service and gRPC server"""
-        await self.server.serve(path=self._address)
+        try:
+            await self.server.serve(log_level=self._log_level, path=self._address)
+        finally:
+            await self.stop()
 
     async def stop(self):
         """Stop the module service and gRPC server"""
+        LOGGER.debug("Shutting down module")
         try:
             if self.parent is not None:
                 await self.parent.close()
-            self.server.close()
```

Comment:
This now means that if `self.parent` is None, it will not close. As long as this is on purpose!

Reply:
Server closure happens automatically now, we don't have to manually call it!